File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.cpp
/* This method is the easiest method to remove duplicates where it directly uses the unique standard template library in C++ to find the duplicate items*/

//more precise and good time complexity

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& number) {
       number.erase(std::unique(number.begin(), number.end()), number.end());
        
        return number.size();
    }
};

File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.java

class Solution {
    public int removeDuplicates(int[] number) {
       int num=number.length;
        if(num==0)
           {
             return num;  
           }
       int k=0;
        for(int m=0; m<=num-1; m++)
        {
           if (number[k] != number[m]) {
				
				k++;
				number[k] = number[m];
			} 
        }
        return k+1;
    }
}